authorization-common/src/main/java/com/sequenceiq/authorization/utils/GetAuthzActionTypeProvider.java

```diff
@@ -0,0 +1,79 @@
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.DATABASE_SERVER;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.DATAHUB;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.DATALAKE;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.ENVIRONMENT;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.FREEIPA;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.IMAGE_CATALOG;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.KERBEROS;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.LDAP;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.PROXY;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.RECIPE;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.STRUCTURED_EVENT;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sequenceiq.authorization.resource.AuthorizationResourceType;
+
+public final class GetAuthzActionTypeProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GetAuthzActionTypeProvider.class);
+
+    private static final Set<AuthzActionTypePair> PAIRS_V2 = new HashSet<>(AuthorizationResourceType.values().length);
+
+    static {
+        PAIRS_V2.add(new AuthzActionTypePair(CREDENTIAL, DESCRIBE_CREDENTIAL));
+        PAIRS_V2.add(new AuthzActionTypePair(DATALAKE, DESCRIBE_DATALAKE));
+        PAIRS_V2.add(new AuthzActionTypePair(ENVIRONMENT, DESCRIBE_ENVIRONMENT));
+        PAIRS_V2.add(new AuthzActionTypePair(DATAHUB, DESCRIBE_DATAHUB));
+        PAIRS_V2.add(new AuthzActionTypePair(IMAGE_CATALOG, DESCRIBE_CREDENTIAL));
+        PAIRS_V2.add(new AuthzActionTypePair(CLUSTER_DEFINITION, DESCRIBE_CLUSTER_DEFINITION));
+        PAIRS_V2.add(new AuthzActionTypePair(CLUSTER_TEMPLATE, DESCRIBE_CLUSTER_TEMPLATE));
+        PAIRS_V2.add(new AuthzActionTypePair(CUSTOM_CONFIGURATIONS, DESCRIBE_CUSTOM_CONFIGS));
+        PAIRS_V2.add(new AuthzActionTypePair(DATABASE, DESCRIBE_DATABASE));
+        PAIRS_V2.add(new AuthzActionTypePair(DATABASE_SERVER, DESCRIBE_DATABASE_SERVER));
+        PAIRS_V2.add(new AuthzActionTypePair(RECIPE, DESCRIBE_RECIPE));
+        PAIRS_V2.add(new AuthzActionTypePair(AUDIT_CREDENTIAL, DESCRIBE_AUDIT_CREDENTIAL, DESCRIBE_ENVIRONMENT));
+        PAIRS_V2.add(new AuthzActionTypePair(STRUCTURED_EVENT, STRUCTURED_EVENTS_READ));
+        PAIRS_V2.add(new AuthzActionTypePair(FREEIPA, DESCRIBE_ENVIRONMENT));
+        PAIRS_V2.add(new AuthzActionTypePair(LDAP, DESCRIBE_ENVIRONMENT));
+        PAIRS_V2.add(new AuthzActionTypePair(PROXY, DESCRIBE_PROXY));
+        PAIRS_V2.add(new AuthzActionTypePair(KERBEROS, DESCRIBE_ENVIRONMENT));
```

Comment:
i don't see the logic for getting envCrn in case of freeipa, ldap and kerberos in order to do proper authorization with DESCRIBE_ENVIRONMENT over envCrn, as discussed

Reply:
as we discussed, since it is more complex to get env crn by freeipa or proxy or kerberos, we should:
- skip authz for these
- log the fact that why we skipped authz (with crn of resource)
- create followup ticket for these cases
- comment the code here with jira id of the followup ticket and with fact about that we should implement more complex logic for these with additional database queries